cep-??.md

```diff
@@ -0,0 +1,246 @@
+>   for example `1.0dev42`, `2.3.5.dev12`.
+> * Post-release versions are indicated by the string `post`,
+>   for example `1.0post1`, `2.3.5.post2`.
+> * Tagged versions have a suffix that specifies a particular
+>   property of interest, e.g. `1.1.parallel`. Tags can be added
+>   to any of the preceding four types. As far as sorting is concerned,
+>   tags are treated like strings in pre-release versions.
+> * An optional local version string separated by `+` can be appended
+>   to the main (upstream) version string. It is only considered
+>   in comparisons when the main versions are equal, but otherwise
+>   handled in exactly the same manner.
+>
+> To obtain a predictable version ordering, it is crucial to keep the
+> version number scheme of a given package consistent over time.
+> Specifically,
+> * version strings should always have the same number of components
+>   (except for an optional tag suffix or local version string),
+> * letters/strings indicating non-release versions should always
+>   occur at the same position.
+> Before comparison, version strings are parsed as follows:
+> * They are first split into epoch, version number, and local version
+>   number at `!` and `+` respectively. If there is no `!`, the epoch is
+>   set to 0. If there is no `+`, the local version is empty.
+> * The version part is then split into components at `.` and `_`.
+> * Each component is split again into runs of numerals and non-numerals
+> * Subcomponents containing only numerals are converted to integers.
+> * Strings are converted to lower case, with special treatment for `dev`
+>   and `post`.
+> * When a component starts with a letter, the fillvalue 0 is inserted
+>   to keep numbers and strings in phase, resulting in `1.1.a1 == 1.1.0a1`.
+> * The same is repeated for the local version part.
+>
+> Examples:
+>
+> ```
+> 1.2g.beta15.rc  =>  [[0], [1], [2, 'g'], [0, 'beta', 15], [0, 'rc']]
+> 1!2.15.1_ALPHA  =>  [[1], [2], [15], [1, '_alpha']]
+> ```
+>
+> The resulting lists are compared lexicographically, where the following
+> rules are applied to each pair of corresponding subcomponents:
+>
+> * integers are compared numerically
+> * strings are compared lexicographically, case-insensitive
+> * strings are smaller than integers, except
+>   * `dev` versions are smaller than all corresponding versions of other types
+>   * `post` versions are greater than all corresponding versions of other types
+> * if a subcomponent has no correspondent, the missing correspondent is
+>   treated as integer 0 to ensure `1.1` == `1.1.0`.
+>
+> The resulting order is:
+>
+> ```
+>    0.4
+>  < 0.4.0
+>  < 0.4.1.rc
+> == 0.4.1.RC   # case-insensitive comparison
+>  < 0.4.1
+>  < 0.5a1
+>  < 0.5b3
+>  < 0.5C1      # case-insensitive comparison
+>  < 0.5
+>  < 0.9.6
+>  < 0.960923
+>  < 1.0
+>  < 1.1dev1    # special case 'dev'
+>  < 1.1_       # appended underscore is special case for openssl-like versions
```

Comment:
I understand this is not part of this CEP but the suffix `_` notion is not present in the description above.. It is also another can of worms. `1.0-` is also valid. So is `1.0__` and `1.0--`...